core/unix/inc/TUnixSigHandling.h

```diff
@@ -0,0 +1,83 @@
+// @(#)root/unix:$Id$
+// Author: Fons Rademakers   15/09/95
+
+/*************************************************************************
+ * Copyright (C) 1995-2000, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TUnixSigHandling
+#define ROOT_TUnixSigHandling
+
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TUnixSigHandling                                                     //
+//                                                                      //
+// Class providing an interface to the UNIX Operating System.           //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef ROOT_TSigHandling
+#include "TSigHandling.h"
+#endif
+#ifndef ROOT_TSysEvtHandler
+#include "TSysEvtHandler.h"
+#endif
+#ifndef ROOT_TTimer
+#include "TTimer.h"
+#endif
+
+typedef void (*SigHandler_t)(ESignals);
+
+
+class TUnixSigHandling : public TSigHandling {
+
+protected:
+   //---- Unix signal interface functions ----------------------
+   static void         UnixSignal(ESignals sig, SigHandler_t h);
+   static const char  *UnixSigname(ESignals sig);
+   static void         UnixSigAlarmInterruptsSyscalls(Bool_t set);
+   static void         UnixResetSignal(ESignals sig);
+   static void         UnixResetSignals();
+   static void         UnixIgnoreSignal(ESignals sig, Bool_t ignore);
+   static void         UnixSetDefaultSignals();
+
+   //---- Unix stack trace helper functions ---------------------
+   static void         StackTraceHelperInit();
+   static void         StackTraceMonitorThread();
+   static void         StackTraceTriggerThread();
+   static void         StackTraceForkThread();
+   static int          StackTraceExecScript(void *);
+   
+public:
+   TUnixSigHandling();
+   virtual ~TUnixSigHandling();
+
+   //---- Misc -------------------------------------------------
+   static const char *Getenv(const char *name);
```

Comment:
Those routine must stay in T[Unix]System as they have not directly related to signal handling.